src/sincosint.jl

```diff
@@ -0,0 +1,232 @@
+const pidiv2 = π/2
+
+import Base.Math: @horner
+
+# This program computes the sine integral ∫_0^x sin(t)/t dt using the rational approximations of A.J. MacLeod, Numer. Algor., 12:259--272, 1996.
+function sinint(x::Float64)
+    t = x*x
+    if abs(x) ≤ 6.0
+        return x * @horner(t, 1.00000000000000000000E0,
+                             -0.44663998931312457298E-1,
+                              0.11209146443112369449E-2,
+                             -0.13276124407928422367E-4,
+                              0.85118014179823463879E-7,
+                             -0.29989314303147656479E-9,
+                              0.55401971660186204711E-12,
+                             -0.42406353433133212926E-15) /
+                   @horner(t, 1.00000000000000000000E0,
+                              0.10891556624243098264E-1,
+                              0.59334456769186835896E-4,
+                              0.21231112954641805908E-6,
+                              0.54747121846510390750E-9,
+                              0.10378561511331814674E-11,
+                              0.13754880327250272679E-14,
+                              0.10223981202236205703E-17)
+    elseif abs(x) ≤ 12.0
+        invt = inv(t)
+        return sign(x)*pidiv2 - cos(x)/x*@horner(invt, 0.99999999962173909991E0,
```

Comment:
You can save a division by combining the `/x` with the polynomial denominator.  i.e. do `cos(x) * @horner(...) / (x * @horner(...))`